framework/keyring-interface.md

```diff
@@ -75,66 +86,65 @@ perform the [Encrypt Data Key](#encrypt-data-key) behavior.
 
 #### Encrypt Data Key
 
-If the [encryption materials](#structures.md#encryption-materials) contain a plaintext data key,
-OnEncrypt MAY encrypt a data key.
-If the encryption materials do not contain a plaintext data key, OnEncrypt MUST NOT encrypt a data key.
+If the input contains a plaintext data key, or if the keyring generates a data key,
+OnEncrypt MAY encrypt the data key.
+Otherwise, OnEncrypt MUST NOT encrypt a data key.
 
-Encrypt Data Key MAY modify the following fields in the [encryption materials](#structures.md#encryption-materials):
+Encrypt Data Key MAY populate the following fields in the [data key materials](#structures.md#data-key-materials):
```

Comment:
"populate" implies that they are empty. What about keyrings that receive input that already contains EDKs? How about "add to"?